### PR TITLE
lib: update open-amp lib to release v2021.10.0

### PR DESCRIPTION
**Pull request by carlocaione**
Origin:
	https://github.com/OpenAMP/open-amp

Commit:
	3c223735ea969199b3c9e2e2a85c9fa7c022b6c0

Status:
	merge open-amp new version after removing the apps dir.

Release Description:
	https://github.com/OpenAMP/open-amp/releases/tag/v2021.10.0

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

### CODE REVIEW

**Comment by hubertmis** [comment]
Is there a manifest update PR in zephyr repo? I cannot find one.

**Comment by carlocaione** [comment]
> Is there a manifest update PR in zephyr repo? I cannot find one.

I thought that should be done after this was merged but I can open a PR for that as well. @galak what's the procedure here?

**Comment by hubertmis** [comment]
I thought the procedure is following:

1. Open PR in dependency repository (open-amp, libmetal, ...)
2. Open manifest update PR in zephyr, pointing to pull/xxxx/head <- this PR runs automatic tests
3. When automatic tests in zephyr PR are green, merge PR in dependency repository
4. Update manifest update PR in zephyr to point to commit hashes of freshly merged commits <- this action re-runs automatic tests
5. When automatic tests in zephyr PR are green, merge PR in zephyr repository

File: open-amp/lib/virtio/virtqueue.c [hunk 1]
```diff
@@ -241,15 +262,24 @@ void *virtqueue_get_available_buffer(struct virtqueue *vq, uint16_t *avail_idx,
...
+	/* Avail.idx is updated by master, invalidate it */
+	VRING_INVALIDATE(vq->vq_ring.avail->idx);
 	if (vq->vq_available_idx == vq->vq_ring.avail->idx) {
 		return NULL;
 	}
 
 	VQUEUE_BUSY(vq);
 
 	head_idx = vq->vq_available_idx++ & (vq->vq_nentries - 1);
+
```
Comment (arnopo) [inline open-amp/lib/virtio/virtqueue.c]: Here there is a delta vs the original open-amp lib. The code below is placed before previous line.

**Review by arnopo — CHANGES_REQUESTED** [review]

File: open-amp/lib/virtio/virtqueue.c (hunk 1 above)
Comment (carlocaione) [reply open-amp/lib/virtio/virtqueue.c]: well, that's embarrassing. I had some local changes in my open-amp repo that were picked up when copying over. Sorry about this.

**Comment by carlocaione** [comment]
> I thought the procedure is following:



https://github.com/zephyrproject-rtos/zephyr/pull/39994

**Review by arnopo — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**